### PR TITLE
minigalaxy: init at 1.1.0

### PR DESCRIPTION
**Pull request by rapenne-s**
###### Motivation for this change

Import minigalaxy https://github.com/sharkwouter/minigalaxy
I reused https://github.com/NixOS/nixpkgs/pull/85658 as a base to make this version, following advice given there.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rapenne-s** [comment]
Using the steam-native-run as explained in https://github.com/NixOS/nixpkgs/pull/85658 just solves everything, minigalaxy runs fine and propose to log-in on GOG and also run the games fine (there are exceptions though like old games requiring i386 libs).

On plasma desktop this program has the issue issue as reported on https://github.com/NixOS/nixpkgs/issues/149812 , I tested with Xfce and it works fine.

File: pkgs/applications/misc/minigalaxy/default.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-BbtwLuG5TH/+06Ez8+mwSAjG1IWg9/3uxzjmgPHczAw=";
+  };
+
+  # Prevent homeless error
+  preCheck = "export HOME=$PWD";
+
+  checkPhase = ''
+    env HOME=$PWD LC_ALL=en_US.UTF-8 pytest
```
Comment (0x4A6F) [inline pkgs/applications/misc/minigalaxy/default.nix:30]:
```suggestion
    runHook preCheck
    env HOME=$PWD LC_ALL=en_US.UTF-8 pytest
    runHook postCheck
```

File: pkgs/applications/misc/minigalaxy/default.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+, glibcLocales
+, glib-networking
+, gobject-introspection
+, gtk3
+, python3
+, python3Packages
+, steam-run-native
+, unzip
+, webkitgtk
+, wrapGAppsHook,}:
```
Comment (0x4A6F) [inline pkgs/applications/misc/minigalaxy/default.nix:13]: Could you format your code with [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt), this looks unfamiliar.

File: pkgs/applications/misc/minigalaxy/default.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      webkitgtk
+  ];
+
+  # Run Linux games using the Steam Runtime by using steam-run in the wrapper
+  postFixup = ''
+    sed -e 's#exec -a "$0"#exec -a "$0" ${steam-run-native}/bin/steam-run#' -i $out/bin/minigalaxy
+  '';
+
+  meta = with lib; {
+    homepage = "https://sharkwouter.github.io/minigalaxy/";
```
Comment (0x4A6F) [inline pkgs/applications/misc/minigalaxy/default.nix:67]:
```suggestion
    homepage = "https://sharkwouter.github.io/minigalaxy/";
    changelog = "https://github.com/sharkwouter/minigalaxy/blob/${version}/CHANGELOG.md";
```

**Review by 0x4A6F — CHANGES_REQUESTED** [review]
Thanks for this port. Looks interesting.

Built and tested on x86_64-linux.

File: pkgs/applications/misc/minigalaxy/default.nix (hunk 2 above)
Comment (0x4A6F) [reply pkgs/applications/misc/minigalaxy/default.nix]: Please also remove the superfluous lines 15, 29-31.

Comment (rapenne-s) [reply pkgs/applications/misc/minigalaxy/default.nix]: thanks !

File: pkgs/applications/misc/minigalaxy/default.nix (hunk 1 above)
Comment (rapenne-s) [reply pkgs/applications/misc/minigalaxy/default.nix:34]: fixed

**Comment by rapenne-s** [comment]
> I wonder if it would be desirable to add wine as a dependency, Minigalaxy can use it to allow installing Windows games but it's an opt-in option that you can only turn On if Wine is found by Minigalaxy.

I tried to add wine support to the package but minigalaxy fails to find wine executable when installing a Windows game. Help would be appreciated if we want to support Windows games (I prefer to use Lutris for those because they require too much fiddling while Linux games with minigalaxy work out of the box)

**Comment by 0x4A6F** [comment]
Result of `nixpkgs-review pr 151136` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>minigalaxy</li>
  </ul>
</details>

**Review by 0x4A6F — APPROVED** [review]

**Pull request merged**